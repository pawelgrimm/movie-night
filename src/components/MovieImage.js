import ReactImageFallback from "react-image-fallback";
import React from "react";
import { getImageUrl } from "../theMovieDb/theMovieDb";

const MovieImage = ({ title, type, imageWidth, filePath, className = "" }) => {
  const imagePath = getImageUrl(type, imageWidth, filePath);
  return (
    <ReactImageFallback
      src={imagePath}
      fallbackImage="/images/default_poster.png"
      initialImage="/images/loader.gif"
      alt={`${type} image for ${title}`}
      className={className || type + "-image"}
    />
  );
};

export default MovieImage;
