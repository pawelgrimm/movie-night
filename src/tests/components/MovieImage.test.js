import { expect } from "@jest/globals";
import { shallow } from "enzyme";
import React from "react";
import MovieImage from "../../components/MovieImage";
import { movies } from "../fixtures/fixtures";

test("should render poster image correctly", () => {
  const movie = movies[2];
  const wrapper = shallow(
    <MovieImage
      title={movie.title}
      filePath={movie.poster_path}
      imageWidth={300}
      type="poster"
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should pass down className correctly", () => {
  const movie = movies[1];
  const wrapper = shallow(
    <MovieImage
      title={movie.title}
      filePath={movie.backdrop_path}
      imageWidth={500}
      type="backdrop"
      className="made-up-class__image"
    />
  );
  expect(wrapper).toMatchSnapshot();
});
