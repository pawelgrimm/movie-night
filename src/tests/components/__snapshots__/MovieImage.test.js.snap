// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass down className correctly 1`] = `
<ReactImageFallback
  alt="backdrop image for Before Sunrise"
  className="made-up-class__image"
  fallbackImage="/images/default_poster.png"
  initialImage="/images/loader.gif"
  src="https://image.tmdb.org/t/p/w780/qA2TyqPldTtoTVY3LKrNIG5g6bH.jpg"
/>
`;

exports[`should render poster image correctly 1`] = `
<ReactImageFallback
  alt="poster image for Before Sunset"
  className="poster-image"
  fallbackImage="/images/default_poster.png"
  initialImage="/images/loader.gif"
  src="https://image.tmdb.org/t/p/w342/gycdE1ARByGQcK4fYR2mgpU6OO.jpg"
/>
`;
